Tests/scripts/infrastructure_tests/common_test.py

```diff
@@ -1,147 +1,124 @@
     """
-    Given a pipeline id, list of pipelines and commits
-    When the pipeline id is not in the list of pipelines
-    Then it should return None, None
+    Given:
+        - Two pipelines with their statuses.
+    When:
+        - Checking on status change. 
+    Then:
+        - It should return the expected result.
+          scenario 1: pipeline1.status =='success' and pipeline2.status == 'failed' -> True
+          scenario 2: pipeline1.status == 'failed' and pipeline2.status =='success' -> False
+          scenario 3: pipeline1.status =='success' and pipeline2.status =='success' -> None
+          scenario 4: pipeline1.status == 'failed' and pipeline2.status == 'in progress' -> None
     """
-    pipeline_id = '1'
-    pipelines = [mocker.Mock(id=2)]
-    commits = [mocker.Mock()]
-
-    expected = (None, None)
+    pipeline1 = mocker.Mock()
+    pipeline1.status = pipeline1_status
+    pipeline2 = mocker.Mock()
+    pipeline2.status = pipeline2_status
 
-    result = common.is_pivot(pipeline_id, pipelines, commits)
+    mocker.patch('Tests.scripts.common.are_pipelines_in_order', return_value=(True))
+    result = is_pivot(pipeline2, pipeline1)
 
     assert result == expected
 
 
-def test_is_pivot_negative(mocker):
+@pytest.mark.parametrize('response, expected', [
+    ([], None),
+    ([{"Jon": "test", "state": "test", "user": {"login": "Jon"}},
+      {"Jane Doe": "test", "state": "APPROVED", "user": {"login": "Jane Doe"}}], "Jane Doe"),
+    ([{"Jon": "test", "state": "APPROVED", "user": {"login": "Jon"}},
+     {"Jane Doe": "test", "state": "APPROVED", "user": {"login": "Jane Doe"}}], "Jon")
+])
+def test_get_reviewer(response, expected, requests_mock: MockerCore):
     """
-    Given a pipeline id, list of pipelines and commits
-    When previous pipeline succeeded and current failed and in order
-    Then it should return True, commit
+    Given:
+        - A PR URL.
+    When:
+        - get_reviewer is called on it.
+    Then:
+        - It should return the expected result.
+        scenario 1: No reviewers -> None
+        scenario 2: One reviewer who approved -> "Jane Doe"
+        scenario 3: Two reviewers who approved -> the first one - "Jon"
     """
-    pipeline_id = '2'
-    pipelines = [
-        mocker.Mock(id=1, status='success'),
-        mocker.Mock(id=2, status='failed')
-    ]
-    commit = mocker.Mock()
-    commits = [commit]
-
-    expected = (True, commit)
-    mocker.patch.object(common, 'are_pipelines_in_order_as_commits', return_value=(True, commit))
-    result = common.is_pivot(pipeline_id, pipelines, commits)
-
+    pr_url = 'https://github.com/owner/repo/pull/123'
+    requests_mock.get('https://api.github.com/repos/owner/repo/pulls/123/reviews', json=response)
+    result = get_reviewer(pr_url)
     assert result == expected
 
 
-def test_is_pivot_positive(mocker):
+@pytest.mark.parametrize('name, expected', [
+    ("Mike", "mike"),
+    ("Jon", "Jon"),
+    ("github-actions[bot]", "the owner")
```

Comment:
add `ids` to the parameterized.
@JasBeilin consider enforcing new parametrize rules.

Reply:
Not furmiliar with that concept, and did not find examples online. Can you please explain what do you mean? in a comma like this?:
#1 ("Mike", "mike"),
#2   ("Jon", "Jon"),

Reply:
This is new for me, cool!